File: examples/python/advanced/serialization/map-parallel-config.py
from aws_durable_execution_sdk_python import DurableContext, durable_execution
from aws_durable_execution_sdk_python.config import MapConfig

@durable_execution
def handler(event: dict, context: DurableContext) -> dict:
    items = [1, 2, 3, 4, 5]
    
    # Custom serialization for BatchResult
    config = MapConfig(
        serdes=CustomSerDes(),        # For the entire BatchResult
        item_serdes=ItemSerDes()      # For individual item results
    )
    
    result = context.map(process_item, items, config=config)
    return {"processed": len(result.succeeded)}

File: examples/java/advanced/serialization/map-parallel-config.java
var items = List.of(1, 2, 3, 4, 5);

// Custom serialization for map results
var config = MapConfig.builder()
    .serDes(new CustomSerDes())          // For the entire MapResult
    .itemSerDes(new ItemSerDes())         // For individual item results
    .build();

var result = ctx.map("process-items", items, Map.class,
    (item, stepCtx) -> processItem(item), config);

return Map.of("processed", result.getSucceeded().size());
